optimism/src/keccak/interpreter.rs

```diff
@@ -165,107 +180,118 @@ pub trait KeccakInterpreter<F: One + Debug + Zero> {
             self.constrain(
                 AbsorbXor(i),
-                self.is_absorb()
-                    * (self.xor_state(i).clone()
-                        - (self.old_state(i).clone() + self.new_state(i).clone())),
+                self.is_absorb(),
+                (self.xor_state(i).clone()
+                    - (self.old_state(i).clone() + self.new_state(i).clone())),
             );
             // In absorb, Check shifts correspond to the decomposition of the new state
             self.constrain(
                 AbsorbShifts(i),
-                self.is_absorb()
-                    * (self.new_state(i).clone()
-                        - Self::from_shifts(&self.vec_sponge_shifts(), Some(i), None, None, None)),
+                self.is_absorb(),
+                (self.new_state(i).clone()
+                    - Self::from_shifts(&self.vec_sponge_shifts(), Some(i), None, None, None)),
             );
         }
     }
 
     /// Constrains 6 checks of padding absorb sponges
-    /// - 1 of degree 3
-    /// - 5 of degree 4
+    /// - 1 of degree 1
+    /// - 5 of degree 2
+    /// Of which:
+    /// - 6 constraints are added only if is_pad() holds
     fn constrain_padding(&mut self) {
         // Check that the padding is located at the end of the message
         let pad_at_end = (0..RATE_IN_BYTES).fold(Self::zero(), |acc, i| {
             acc * Self::two() + self.in_padding(i)
         });
         self.constrain(
             PadAtEnd,
-            self.is_pad() * (self.two_to_pad() - Self::one() - pad_at_end),
+            self.is_pad(),
+            (self.two_to_pad() - Self::one() - pad_at_end),
         );
         // Check that the padding value is correct
         for i in 0..PAD_SUFFIX_LEN {
             self.constrain(
                 PaddingSuffix(i),
-                self.is_pad() * (self.block_in_padding(i) - self.pad_suffix(i)),
+                self.is_pad(),
+                (self.block_in_padding(i) - self.pad_suffix(i)),
             );
         }
     }
 
     /// Constrains 16 checks of squeeze sponges
-    /// - 16 of degree 2
+    /// - 16 of degree 1
+    /// Of which:
+    /// - 16 constraints are added only if is_squeeze() holds
     fn constrain_squeeze(&mut self) {
         let sponge_shifts = self.vec_sponge_shifts();
         for i in 0..QUARTERS * WORDS_IN_HASH {
             // In squeeze, check shifts correspond to the 256-bit prefix digest of the old state (current)
             self.constrain(
                 SqueezeShifts(i),
-                self.is_squeeze()
-                    * (self.old_state(i).clone()
-                        - Self::from_shifts(&sponge_shifts, Some(i), None, None, None)),
+                self.is_squeeze(),
+                (self.old_state(i).clone()
+                    - Self::from_shifts(&sponge_shifts, Some(i), None, None, None)),
             );
         }
     }
 
     /// Constrains 389 checks of round steps
     /// - 384 constraints of degree 3
```

Comment:
Do you need to update these comments, or the degree stays the same?

Reply:
oh you're right, they are outdated in those comments. Will push a commit